artemis-server/src/main/java/org/apache/activemq/artemis/core/deployers/impl/FileConfigurationParser.java

```diff
@@ -768,6 +772,45 @@ public void parseMainConfig(final Element e, final Configuration config) throws
       }
    }
 
+
+   private void parseJournalRetention(final Element e, final Configuration config) {
+      NodeList retention = e.getElementsByTagName("journal-retention");
+
+      if (retention.getLength() != 0) {
+         Element node = (Element) retention.item(0);
+
+         String directory = getAttributeValue(node, "directory");
+         long storageLimit = ByteUtil.convertTextBytes(getAttributeValue(node, "storage-limit").trim());
+         int period = getAttributeInteger(node, "period", 0, Validators.GT_ZERO);
+         String unitStr = getAttributeValue(node, "unit");
+
+         TimeUnit unit;
+
+         switch (unitStr) {
+            case "DAYS":
+               unit = TimeUnit.DAYS;
+               break;
+            case "HOURS":
+               unit = TimeUnit.HOURS;
+               break;
+            case "MINUTES":
+               unit = TimeUnit.MINUTES;
+               break;
+            case "SECONDS":
+               unit = TimeUnit.SECONDS;
+               break;
+            default:
+               unit = TimeUnit.DAYS;
+         }
+
+         config.setJournalRetentionDirectory(directory);
+         config.setJournalRetentionTimeUnit(unit);
```

Comment:
So that two args throughout code base dont have to deal with time conversion and also in more critical code it doesnt have to be converted later why not convert to ms at point of config load

Reply:
I will think about this between tomorrow (Saturday) and Monday.